src/MockCollection.php

```diff
@@ -49,7 +49,14 @@ public function insertOne($document, array $options = [])
             $document = new BSONDocument($document);
         }
 
-        $document = new BSONDocument($document);
+        // Possible double instantiation of BSONDocument?
+        // With or without the below, it seems that the
+        // BSONDocument class recurses automatically.
+        // e.g. $document->bsonSerialize() will give the same result
+        // so I couldn't write a test to capture the need or not for this.
+        // I'm commenting it out anyway.
+        // $document = new BSONDocument($document);
```

Comment:
Good catch; you actually *can* initialize a `BSONDocument` with another one, and it will automatically unbox the value. It doesn't hurt, but also doesn't do any good. I'd be fine with removing this statement altogether.